src/Display/Modal/Modal.tsx

```diff
@@ -58,6 +57,26 @@ class Modal extends React.Component<Props, State> {
     document.body.removeAttribute('style');
   }
 
+  // To prevent the modal from closing
+  // when a mousedown event occurs inside the ModalContentArea
+  // but the subsequent mouseup event occurs outside
+  handleMouseDown = (e: React.MouseEvent) => {
+    const element = event.target as HTMLElement;
+    const modalContentAreaNode = this.modalContentAreaRef.current;
+    const isClickInsideModal = modalContentAreaNode.contains(element);
+    if (isClickInsideModal) {
+      this.setState({ ignoreBackdropClick: true });
+    }
+  };
+
+  handleClick = () => {
+    if (this.state.ignoreBackdropClick) {
+      this.setState({ ignoreBackdropClick: false });
+      return;
+    }
+    this.props.onClose();
+  };
```

Comment:
@franciscrispin  Could we use below to instead `ignoreBackdropClick` simply?
```
  handleMouseDown = (e: React.MouseEvent) => {
    const element = event.target as HTMLElement;
    this.mouseDownTarget = element;
  };

  handleClick = (e: React.MouseEvent) => {
    const element = event.target as HTMLElement;

    if (this.mouseDownTarget === element) {
      this.props.onClose();
    }
  };
```

Reply:
Oh yes this is great! Thanks @rosslo. PTAL again :smile: